bookOfHb.md

```diff
@@ -0,0 +1,400 @@
+# The Book of Self-Repair (Heartbeat First Impressions)
+
+##  Overall Goals of Self-Repair
+
+1.  Host self-repair recipes, outside of firefox core.  Run them with Firefox permissions.  Allow fast development off the trains.
+
+2.  Collect randomly sampled User Sentiment scores.  Implemented as "Heartbeat First Impressions" recipe.
+
+
+## As a developer, I want to successfully hack on Self-Repair
+
+### Setup (do once)
+
+1.  Zen Of Cooking
+
+	This is the short version. `Cmd+F` will find bigger explanations of nearly everything later on in this document.  
+
+   Mastering this may take several attempts.  Skip around.  Be gentle on yourself!  Enjoy the process!
+   
+1. Pots and Pans
+
+	```
+	# required: git, node
+	brew install node   # includes npm
+	brew install git    # git
+	
+	# optional:  hub, git
+	brew install hub    # github's replacement git client
+	brew install ghi    # github issues client
+	```
+
+1. Groceries
+
+	```
+	# usual git project clone and fork
+	
+	git clone mozilla/self-repair-server
+	cd self-repair-server
+	hub fork
+	```
+
+1. Prep the veggies
+
+	```
+	# usual node project package install
+	npm install   
```

Comment:
Ok. I learned a lot about node.js and npm just now. I've got:

* node v4.1.0
* npm v2.14.3

That combination works super. The versions that come in the Fedora 22 repositories are:

* node v0.10.36
* npm 1.3.6

Those don't work at all. Hope that helps your digging.